### PR TITLE
Downloading translations and ignoring branch at Crowdin

### PR DESCRIPTION
**Pull request by jonasfranz**
This PR adds the following build steps:

* download_translations: Downloads & exports all translations from crowdin and saves them into options/locale/
* upload_translations: Uploads the updated locale_en-US.ini to Crowdin

The build steps are splitted because the files should only uploaded to crowdin if all test passes.

It also adds the `ignore_branch` option to fix the build process because gitea does not use branches at crowdin.

This is already tested here:
* Upload & Export in custom folder: http://drone.jonasfranz.software/JonasFranzDEV/Crowdin-Demo/12
* Used configuration: https://git.jonasfranz.software/JonasFranzDEV/Crowdin-Demo/src/master/.drone.yml

Related to #2585
Related to JonasFranzDEV/drone-crowdin#2

### CODE REVIEW

**Comment by lafriks** [comment]
I don't quite see point in downloading translations from crowdin. As we need to get these translations in git

**Comment by lunny** [comment]
Two options, one is manually download crowdin languages packs and send a PR, another is remove all language packs except en-US and download it before build.

**Comment by jonasfranz** [comment]
@lunny @lafriks I think that we need a combination out of both. So if you release a new version you should download the translations manually. If you build a master version you will get the translations on the fly via the build process.

**Comment by lafriks** [comment]
@JonasFranzDEV I like @tboerger idea that appleboy/drone-git-push could be used to push downloaded files to git

**Comment by tboerger** [comment]
> @JonasFranzDEV I like @tboerger idea that appleboy/drone-git-push could be used to push downloaded files to git

In reference to https://github.com/go-gitea/gitea/pull/2585#issuecomment-331798414

**Comment by jonasfranz** [comment]
@tboerger @lafriks I've added the git push part. The translations will be updated on GitHub too now.

File: .drone.yml [hunk 1]
```diff
@@ -20,6 +20,18 @@ pipeline:
     when:
       event: [ push ]
       branch: [ master ]
+  git_push:
+    image: appleboy/drone-git-push
+    pull: true
+    secrets: [ git_push_ssh_key ]
+    remote: "git@github.com:go-gitea/gitea.git"
```
Comment (tboerger) [inline .drone.yml]: I think we don't need the remote

File: .drone.yml [hunk 2]
```diff
@@ -9,6 +9,31 @@ clone:
...
+    when:
+      event: [ push ]
+      branch: [ master ]
+      
+  git_push:
+    image: appleboy/drone-git-push
+    pull: true
+    secrets: [ git_push_ssh_key ]
+    remote: "git@github.com:go-gitea/gitea.git"
+    branch: master
```
Comment (tboerger) [inline .drone.yml]: The branch should be already master?

File: .drone.yml [hunk 3]
```diff
@@ -211,8 +235,16 @@ pipeline:
     when:
       event: [ push ]
       branch: [ master ]
-
-  translations:
+      
+  github:
```
Comment (lunny) [inline .drone.yml]: This section seems not related?

Comment (lafriks) [reply .drone.yml]: @JonasFranzDEV why this section is needed here?

Comment (daviian) [reply .drone.yml]: @lunny @lafriks As you can see in comments: this was a merge from master to pr branch. Think a full rebase is necessary.

Comment (jonasfranz) [reply .drone.yml]: @daviian What should I do for a "full rebase"?

Comment (Morlinest) [reply .drone.yml]: `git rebase master`?

**Comment by jonasfranz** [comment]
I've fully rebased it onto master. @Morlinest

**Comment by Morlinest** [comment]
I think something is wrong, 51 files changed.

File: .drone.yml (hunk 3 above)
Comment (daviian) [reply .drone.yml]: @JonasFranzDEV Usually I rebase my master to be the same as upstream/master (original gitea remote) and after checking out my pr branch I do it like @Morlinest said.

**Comment by jonasfranz** [comment]
Replaced by #2727
